ewhunting.py

```diff
@@ -0,0 +1,1768 @@
+                cursor.close()
+                ewutils.databaseClose(conn_info)
+
+    """ Save enemy data object to the database. """
+
+    def persist(self):
+        try:
+            conn_info = ewutils.databaseConnect()
+            conn = conn_info.get('conn')
+            cursor = conn.cursor();
+
+            # Save the object.
+            cursor.execute(
+                "REPLACE INTO enemies({}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}, {}) VALUES(%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)".format(
+                    ewcfg.col_id_enemy,
+                    ewcfg.col_id_server,
+                    ewcfg.col_enemy_slimes,
+                    ewcfg.col_enemy_totaldamage,
+                    ewcfg.col_enemy_ai,
+                    ewcfg.col_enemy_type,
+                    ewcfg.col_enemy_attacktype,
+                    ewcfg.col_enemy_display_name,
+                    ewcfg.col_enemy_identifier,
+                    ewcfg.col_enemy_level,
+                    ewcfg.col_enemy_poi,
+                    ewcfg.col_enemy_life_state,
+                    ewcfg.col_enemy_bleed_storage,
+                    ewcfg.col_enemy_time_lastenter,
+                    ewcfg.col_enemy_initialslimes,
+                    ewcfg.col_enemy_lifetime,
+                    ewcfg.col_enemy_id_target,
+                    ewcfg.col_enemy_raidtimer,
+                ), (
+                    self.id_enemy,
+                    self.id_server,
+                    self.slimes,
+                    self.totaldamage,
+                    self.ai,
+                    self.type,
+                    self.attacktype,
+                    self.display_name,
+                    self.identifier,
+                    self.level,
+                    self.poi,
+                    self.life_state,
+                    self.bleed_storage,
+                    self.time_lastenter,
+                    self.initialslimes,
+                    self.lifetime,
+                    self.id_target,
+                    self.raidtimer,
+                ))
+
+            conn.commit()
+        finally:
+            # Clean up the database handles.
+            cursor.close()
+            ewutils.databaseClose(conn_info)
+
+    # Function that enemies use to attack or otherwise interact with players.
+    async def kill(self):
+
+        client = ewutils.get_client()
+
+        last_messages = []
+        should_post_resp_cont = True
+
+        enemy_data = self
+
+        time_now = int(time.time())
+        resp_cont = ewutils.EwResponseContainer(id_server=enemy_data.id_server)
+        district_data = EwDistrict(district=enemy_data.poi, id_server=enemy_data.id_server)
+        market_data = EwMarket(id_server=enemy_data.id_server)
+        ch_name = ewcfg.id_to_poi.get(enemy_data.poi).channel
+
+        target_data = None
+        target_player = None
+        target_slimeoid = None
+
+        attacktype = 'unarmed'
```

Comment:
any string or int that is used to identify a game object should be saved as a constant in ewcfg and only referenced through that constant. particularly strings invite typos when repeatedly typed out like this and generally make future maintenance more of a hassle.